README.md

```diff
@@ -1,2 +1,187 @@
-# py-composite-action-lib
-PyCompositeActionLib is a Python library that streamlines creating and managing composite GitHub Actions. It offers reusable classes and scripts for controlling action inputs/outputs and accessing the GitHub API. 
+# PyCompositeActionLib
+
+- [Description](#description)
+- [Motivation](#motivation)
+- [How to Use](#how-to-use)
+  - [Example: Control Action Inputs and Outputs](#example-control-action-inputs-and-outputs)
+  - [Example: Access GitHub API with GithubManager](#example-access-github-api-with-githubmanager)
+- [Features](#features)
+- [How to Use Local Version of Library for Development](#how-to-use-local-version-of-library-for-development)
+- [How to Run Tests](#how-to-run-tests)
+- [How to Release](#how-to-release)
+- [Contribution Guidelines](#contribution-guidelines)
+- [License Information](#license-information)
+- [Contact or Support Information](#contact-or-support-information)
+
+
+## Description
+**PyCompositeActionLib** is a Python library that streamlines creating and managing composite GitHub Actions. It offers
+reusable classes and scripts for controlling action inputs/outputs and accessing the GitHub API. 
+
+## Motivation
+The motivation behind PyCompositeActionLib is to provide developers with a robust set of tools that simplify
+the creation of composite GitHub Actions. By leveraging reusable components and streamlined API access, this library
+aims to reduce redundancy, improve efficiency, and enhance the overall development experience for creating complex
+workflows on GitHub.
+
+## How to use
+### Example: Control Action Inputs and Outputs
+```python
+from py_composite_action_lib import get_action_input, set_action_output, set_action_failed
+
+# Example: Get action input
+repo_name = get_action_input('repo-name')
+if not repo_name:
+    set_action_failed('The input "repo-name" is required.')
+
+# Example: Set action output
+set_action_output('output-name', 'success')
+```
+
+### Example: Access GitHub API with GithubManager
```

Comment:
how do you access ref?

Reply:
Access to ref is not needed in planned actions.
- The **release notes generator** scans the entire repository using GH REST API.
- The **living documentation generator** scans multiple repositories and projects using GH REST API.
- The **Filename inspector** - scans the current filesystem using provided paths and glob logic.